packages/react-query/src/HydrationBoundary.tsx

```diff
@@ -3,7 +3,11 @@ import * as React from 'react'
 
 import { hydrate } from '@tanstack/query-core'
 import { useQueryClient } from './QueryClientProvider'
-import type { HydrateOptions, QueryClient } from '@tanstack/query-core'
+import type {
+  DehydratedState,
+  HydrateOptions,
+  QueryClient,
+} from '@tanstack/query-core'
 
 export interface HydrationBoundaryProps {
```

Comment:
`HydrationBoundaryProps` don't seem to have changed, which means I can still pass `options.defaultOptions.mutations`, which doesn't make much sense.

I think we could `omit` this on type level

Reply:
Nice catch, should be fixed now.